src/java.base/share/classes/java/lang/VirtualThread.java

```diff
@@ -1444,40 +1448,54 @@ private static ForkJoinPool createDefaultScheduler() {
     /**
      * Schedule a runnable task to run after a delay.
      */
-    private static Future<?> schedule(Runnable command, long delay, TimeUnit unit) {
-        long tid = Thread.currentThread().threadId();
-        int index = (int) tid & (DELAYED_TASK_SCHEDULERS.length - 1);
-        return DELAYED_TASK_SCHEDULERS[index].schedule(command, delay, unit);
+    private Future<?> schedule(Runnable command, long delay, TimeUnit unit) {
+        if (scheduler instanceof ForkJoinPool pool) {
+            return pool.schedule(command, delay, unit);
+        } else {
+            return DelayedTaskSchedulers.schedule(command, delay, unit);
```

Comment:
@AlanBateman Would it make sense to test if the Scheduler implements ScheduledExecutorService?

Reply:
Not for now. If a custom scheduler feature is exposed some time then we can think about this topic, it may or may be that the custom scheduler supports delayed tasks.